### DISCUSSION TITLE
Mid frame AXIS pausing+backpressuring

### DISCUSSION THREAD

**Discussion by OCElephant** [alexforencich/cocotbext-axi#67] [General]
From my understanding, it is not possible to pause the transmission of data mid frame (deassert tvalid in the middle of transmitting an axis frame) with `AxiStreamSource`. Likewise for `AxiStreamSink`, I haven't found a way to exert backpressure mid frame.

I have tried the `set_pause_generator()` function to pause transmission and exert back pressure (for source and sink respectively), however this only seems to pause between frames.

Is mid frame pausing/backpressuring possible? Thanks.

**Comment by alexforencich** [comment]
So, this should be possible, but I think I just found a bug in how the pausing is handled that results in incorrect behavior under certain conditions.  It's possible that you ran in to this bug; I'll get it fixed soon.
